AvicBot
===

(currently inactive)


(when me and abby went over the first 5 revisions, rev id's: 876773821, 877630206, 878854297, 878872180, 877568508)
* on AfCCSD page (what does that mean?), adds a user for empty page, later returns page to empty state

* on same page, adds draft one edit, removes 1 hour later

* on AvicBot/speedy page, adding and deleting various pages, categories, and drafts. unsure about the purpose of this list

* on AfCCSD page, empties page and empties again later

* adding and removing drafts from AfCCSD page.

note: looked at these pages and are no longer in use as of July 2020


do a write-up where we go deeper on some pairs - pick 2 self reverting and 1 bot-bot reverting combination -- each take one bot. can do in spreadsheet or notebook. go beyond bullet pointed list -- write a summary of why bot exists, what it does, why it gets reverted. give us a case study. what is the thing that is trying to be done here? deepen qualitative understanding

* why exist
* how long operational
* what percentage reverts
* why reverts itself
* namespaces
* link to code of bot? -- is publicly accessible?
* how does it work
* look at 10, 20 revisions
* look at 10, 20 reverted revisions

What does this bot do? How long has it been operational? Look at all the revisions this bot makes (10-20 revisions) and then 10-20 of its reverted revisions. What percentage of bots does it revert the most and get reverted the most? How does it interact with reverts? Go to talk pages 
Can’t find the code? (interesting to note) 
Require an understanding of the system

points not in order. collect as many links as you can. if find interesting rev, copy url and write why interesting. klepto approach - take as many as you can


Why AvicBot exist?
==

https://en.wikipedia.org/wiki/User:AvicBot

Source code is published, check the link above^^

AvicBots taks include: 
* maintaining interiwki links
* fixing redirects
* tagging uncategorized pages - adding a template that will add tagged pages to Category:Uncategorized pages or dated subcategories thereof.
* maintian a list of some categories - click the link above for exactly which caategories
* "category multiple merge targets" - you can merge multiple articles together and the target article is the one who oyou want to merge to - seems to be moving and nominating certian categories to this page, which are pages that "need attention and work by editors before the CfD request can be completed. - a similar process to the "articles for deletion" because a discussion is held determining whehter two or more articles shoud be merged
* maintaining {commonscat} category - used on roughly one percent of pages, this template {{Commons category}} links to a "Category:" name in name in the Wikimedia Commons repository that corresponds to the article or category.  If the Commons category name is misspelled or missing, the link will still appear "blue" even though the link will fail
* maintaining indef-blocked categories - user pages in this category are for users who have been indefinetely blocked and now oonly exist temporarily, usually to provide information to the users or allow them a suitable period of time to contest blocking. Indefinite does not mean "infinite" or "permanent". An indefinitely blocked user may later be unblocked in appropriate circumstances. In particularly serious cases in which no administrator would be willing to lift the block, the user is effectively banned by the community.
* tagging {{orphan}} pages - Template:Orphan is used on pages with no incoming links from other articles or lists (orphan). The template is responsible for populating sub-categories of Category:Orphaned articles ( a hidden category). There are currently 91,740 orphaned articles.
* merging 3+ {{multiple issues}} tags - This template tags articles or sections for multiple maintenance issues and then displays the alerts in a single box, as an alternative to using several space-consuming boxes, makes it mroe pleasant to look at, saves space.
* assisting with WP:CFD/W - CfD  is the central venue for discussing specific proposals to delete, merge, rename or split categories and stub types in accordance with the guidelines for categorization, category naming and stub articles.
* Maintaining CAT: UWCOI - "category: User talk pages with conflict of interest notcies" - this is a tracking category - It is used to build and maintain lists of pages. This category lists user talk pages that have received either a {{subst:Uw-coi}} or {{subst:Uw-coi-username}} notice. This information is used for spam and conflict of interests tracking. Talk pages are added automatically to this category via transclusion from either of the templates previously noted.

How long did AvicBot exist? 
==

the REquests for approval for AvicBot was filed on February 7 2011, however the result of the discussion was DENIED (https://en.wikipedia.org/wiki/Wikipedia:Bots/Requests_for_approval/AvicBot) and the fuinciton overview was to remove test edits and it eas denied because "This doesn't seem like a good task for a bot, or for an editor at bot-like speeds. The presence of such links in the article is a good sign that the history needs to be checked for more vandalism, not just "if the vandalism is right around the file". I do encourage you to go through these links and do the history checking to truly clean up the articles." There are multiple AvicBots though

Seems like most of the discussion for the request of bot approval went through a trial process for a certain number of edits, and once, the trial is complete, people leave their comments a member of the Bots Approval Group gives it the go ahead. Sometimes the results of the trial are posted to talk pages. Most of the edits that the trial is for is VERY small (all <60 edits)!!


* it seems that the number next to it doesn't represent a completely idfferent bot. It reperesents the discussion number for the requests of bot approval (or every time a bot wants/has another task 

 * 1. 2/7/2011 - DENIED - function overview: remove test edits - "This doesn't seem like a good task for a bot, or for an editor at bot-like speeds. The presence of such links in the article is a good sign that the history needs to be checked for more vandalism, not just 'if the vandalism is right around the file.'"
 * 2. 4/23/2011 - APPROVED - function overiew: find one category and replaces it with two, as needed; CATEGORY MULTIPLE MERGE TARGETS - 
 * 3. 7/3/2011 - APPROVED - function overview: standard interwiki bot, using pywikipedia; maintaing interwiki links- 
 * 4. 7/22/2011 (3:22) - WITHDRAWN BY OPERATOR - function overview: chaning links, update website links to new site, as there are no redirects, so all the old links are busted - withdrawn becuase "another user has taken up the task with AWB themselves." by Tamuz and closed by HELLKNOWZ
 * 5. 7/22/2011 (3:43PM) - APPROVED - function overview adding Cat interwikis/fixing double redirects; Maintain interwikis for category namespace, and fixing double redirects - "Looks good. Standard non-controversial tasks done by many other bots" by HELLKNOWZ (a member of Bot Approvals Group, an AfC reviewer ~ Articles of Creation.)
 * 6. 7/29/2011 - APPROVED - function overview: mainatin indef blocked categories; "Will grab users in the indef block categories, and check to see if they are, in fact, indef blocked. If a user is not found to be blocked, the username will be set aside on a subpage. If, after two weeks, the user is still in the category, and still unblocked, AvicBot will remove the category from the user talk page."
 * 7. 8/17/2011 - APPROVED - function overview: maintaining (and adding?) {{Commonscat}} templates; further details about the fucntion on this page: https://en.wikipedia.org/wiki/Wikipedia:Bots/Requests_for_approval/AvicBot_7
 * 8. 8/18/2011 - APPROVED - function overview: tagging uncategorized articles; "The bot will scan for Untagged Uncategorized Articles, and add {{Uncategorized|date={{subst:CURRENTMONTHNAME}} {{subst:CURRENTYEAR}}}} to them"
 * 9. 9/8/2011 - APPROVED - function overview: tag orphan pages: " Scans over pages in mainspace and finds orphans (Zero incoming links). If an Orphan page is found, it checks to see if the orphan tag (or "multiple issues|orphan=date" tags ) are on the page. It automatically skips redirects, {{wi}} pages, and disambig pages."
 * 10. 11/20/2011 - APPROVED - funciton overview: groupl multiple issues tags if there are 3 or more eligible tags; same logic as the AvicBot 9 task
 * 11. 5/27/2012 - APPROVED - function overview: performing category (re)moves as listed on WP:CFD/W; "AvicBot will either 1.) move pages from [[Caregory:foo]] to [[Category:bar]], as listed on Wikipedia:Categories for discussion/Working, if it's a move/merge request, or 2.) remove [[Caregory:foo]] entirely if the requests is to empty the category. Note that as AvicBot is not an AdminBot, it will not be deleting any pages." (SEEMS LIKE SOMETHING THAT WOULD CAUSE REVERTS OF ITS OWN EDITS). 
 * 12. 10/7/2015 - APPROVED - function overview: maintainign Category:User talk pages with conflict of interestr notices. Removing category after 6 months; "This will help to remove 'stale' reports and keep the category up to date and userful." (AN EXAMPLE OF A SELF-REVERT THAT MOST LIKELY NOT HAPPEN WITHIN 24 HORUS) 
 
 

How does it interact with reverts?
==

When lookin at the User contoributions for AvicBot, I put filtered the results to only show contributions from Jaunaury 1-21 in 2019, and when briefy scrollign thorugh, AvicBot has the SAME edit summaries: Bot: Listifying from Category:Candidates for speedy deletion, Candidates for speedy deletion as abandoned drafts or AfC submissions, Wikipedia usernames with possible policy issues, User talk pages with conflict of interest notices, Categories with incorrect underpopulated category template (which had the tag: blanking, which is most likely to cause a self-revert, if you're clearing the page), All articles proposed for deletion, AfC postponed G13, G13 eligible AfC submissions, Wikipedians who are indefinitely blocked for advertising

* Seems like AvicBot works a lot in tracking categories, but then again of course we don't know if these are all reverts

In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# other imports can go here
from datetime import datetime
import pytz
import tqdm

In [3]:
df = pd.read_csv('revs_scored_jan.tsv', sep='\t', header=0)

In [4]:
days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [5]:
# first thing: isolate data to only data from Jan1-Jan21, reverted w/in 24 hours, revisions by a bot

sdf = df[df.day<22]
sdf_reverted = sdf[sdf.is_reverted] # w/o, still include 'none'
sdf_reverted_in_day = sdf_reverted[sdf_reverted.seconds_to_revert.astype('str').astype('int')<86400]
sdf_bots = sdf_reverted_in_day[sdf_reverted_in_day.user_is_bot]
# sdf_bots = revisions from week 1-3 which were reverted within 24 hours

In [6]:
# to do next: find which in df_bots match rev_id of where user_is_bot is true?
sdf_another = df[df.user_is_bot]
sdf_another = sdf_another[sdf_another.is_revert]
#sdf_another = all of the bots that reverted a edit/revision

In [7]:
# sdf_bots holds reverted stuff within 3 weeks by bots
sdf1 = sdf_bots.loc[sdf_bots.is_reverted, ["rev_id", "user_text", "revert_id"]]
sdf1.revert_id = sdf1.revert_id.astype('int')

# sdf_another holds possible reverting changes
sdf2 = sdf_another[["rev_id", 'user_text']]

In [8]:
# subset bot-bot reverts using merge

bot_bot_reverts = pd.merge(sdf1, sdf2, 
             how='inner', 
             left_on='revert_id', 
             right_on='rev_id', 
             suffixes=('', '_reverter')
)

In [9]:
bot_pairs = bot_bot_reverts.groupby(["user_text","user_text_reverter"], as_index = False).count().sort_values("rev_id", ascending=False).reset_index()

In [10]:
bot_pairs["is_self_revert"] = False

for row in bot_pairs.itertuples():
    if row.user_text_reverter == row.user_text:
        bot_pairs.loc[row.Index, "is_self_revert"] = True
        

bot_pairs

index                   user_text      user_text_reverter  rev_id  \
0      24          HBC AIV helperbot5      HBC AIV helperbot5    1185   
1      11                  Cyberbot I              Cyberbot I    1004   
2      13                     Cydebot                 Cydebot     820   
3       5                     AvicBot                 AvicBot     532   
4      19                    EmausBot                EmausBot     360   
5      27                 ListeriaBot             ListeriaBot     298   
6      16                      DatBot      HBC AIV helperbot5     295   
7      17                DeltaQuadBot            DeltaQuadBot     136   
8       7                 ClueBot III             ClueBot III     107   
9      36                 Reports bot             Reports bot      92   
10     34                    MusikBot                MusikBot      78   
11     35                    RMCD bot                RMCD bot      63   
12     28       Lowercase sigmabot II              Cyberbot I      61   
13     26                     Legobot                 Legobot      45   
14      6                      COIBot                  COIBot      31   
15     29       Lowercase sigmabot II              Hazard-Bot      26   
16     41                WOSlinkerBot            WOSlinkerBot      21   
17      1                      Acebot                  Acebot      20   
18      3                   AnomieBOT               AnomieBOT      19   
19     14                     DPL bot                 DPL bot      17   
20     15                DYKUpdateBot            DYKUpdateBot      14   
21      2              Amalthea (bot)          Amalthea (bot)      12   
22     31      Lowercase sigmabot III  Lowercase sigmabot III      11   
23     22             Filedelinkerbot         Filedelinkerbot       8   
24     20              EnterpriseyBot          EnterpriseyBot       7   
25     12                 Cyberbot II             Cyberbot II       6   
26     30       Lowercase sigmabot II   Lowercase sigmabot II       4   
27      0                   AAlertBot               AAlertBot       4   
28     32                     Mathbot                 Mathbot       3   
29     33  MediaWiki message delivery             ClueBot III       2   
30      9             CommonsDelinker         CommonsDelinker       2   
31      4                   AnomieBOT                RMCD bot       2   
32     37                      RonBot                  RonBot       2   
33     39                     SineBot      HBC AIV helperbot5       2   
34     40                     SineBot              Hazard-Bot       2   
35     25              JCW-CleanerBot          JCW-CleanerBot       1   
36     23                  GreenC bot              GreenC bot       1   
37     18                    EmausBot                 Bot1058       1   
38     10          Community Tech bot      Community Tech bot       1   
39      8                  ClueBot NG                EmausBot       1   
40     38                     RussBot              Hazard-Bot       1   
41     21                   Femto Bot               Femto Bot       1   

    revert_id  rev_id_reverter  is_self_revert  
0        1185             1185            True  
1        1004             1004            True  
2         820              820            True  
3         532              532            True  
4         360              360            True  
5         298              298            True  
6         295              295           False  
7         136              136            True  
8         107              107            True  
9          92               92            True  
10         78               78            True  
11         63               63            True  
12         61               61           False  
13         45               45            True  
14         31               31            True  
15         26               26           False  
16         21               21 

In [11]:
bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"] #ALL SELF-REVERTS

rev_id user_text  revert_id  rev_id_reverter user_text_reverter
46    876243659   AvicBot  876271234        876271234            AvicBot
47    876244058   AvicBot  876271234        876271234            AvicBot
48    876244450   AvicBot  876271234        876271234            AvicBot
49    876244893   AvicBot  876271234        876271234            AvicBot
50    876245314   AvicBot  876271234        876271234            AvicBot
...         ...       ...        ...              ...                ...
5244  879508548   AvicBot  879511314        879511314            AvicBot
5245  879510573   AvicBot  879511314        879511314            AvicBot
5269  879511290   AvicBot  879518555        879518555            AvicBot
5270  879514610   AvicBot  879519188        879519188            AvicBot
5282  879530797   AvicBot  879533764        879533764            AvicBot

[532 rows x 5 columns]

Now exploirng TOP 20 reverted and "reverting" edits 
==


In [12]:
bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"].head(20)

rev_id user_text  revert_id  rev_id_reverter user_text_reverter
46   876243659   AvicBot  876271234        876271234            AvicBot
47   876244058   AvicBot  876271234        876271234            AvicBot
48   876244450   AvicBot  876271234        876271234            AvicBot
49   876244893   AvicBot  876271234        876271234            AvicBot
50   876245314   AvicBot  876271234        876271234            AvicBot
51   876254649   AvicBot  876271234        876271234            AvicBot
135  876267825   AvicBot  876268182        876268182            AvicBot
151  876270552   AvicBot  876277467        876277467            AvicBot
152  876270563   AvicBot  876275873        876275873            AvicBot
235  876286171   AvicBot  876289513        876289513            AvicBot
236  876287145   AvicBot  876289513        876289513            AvicBot
237  876287797   AvicBot  876289513        876289513            AvicBot
238  876289121   AvicBot  876289513        876289513            AvicBot
258  876289513   AvicBot  876295082        876295082            AvicBot
259  876292591   AvicBot  876295082        876295082            AvicBot
260  876293533   AvicBot  876295082        876295082            AvicBot
261  876294173   AvicBot  876295082        876295082            AvicBot
294  876300057   AvicBot  876300758        876300758            AvicBot
309  876308853   AvicBot  876314649        876314649            AvicBot
310  876309523   AvicBot  876314649        876314649            AvicBot

In [13]:
Acivbot_self_reverts = bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"]

THESE ARE THE FIRST 20 REVISIONS WITHIN THESE THREE WEEKS!


* rev id: 876243659 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; this revert is just showed when the page was back to 0 entries 

* rev id: 876244058 - This edit summary was the same as the previous one, yet the revisions don't seem to be related and don't even seem to be reverts... 

*  rev id: 876244450 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty)

*  rev id: 876244893 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty), 9 to 13 to 0 entries

*  rev id: 876245314 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 13 to 19 to 0

*  rev id: 876254649 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 19 to 1 to 0

*  rev id: 876267825 - This edit summary was "Listifying from Category:Candidates for speedy deletion". A user was added and that same user was deleted, exact self revert

*  rev id: 876270552 -  This edit summary was "Listifying from Category:Wikipedia usernames with possible policy issues". A user was added and two user talk pages switched positions, then  that same user was deleted and those same user talk pages switched backed to their orignal positions, exact self revert

*  rev id: 876270563 - This edit summary was the same as the last one. The revision was the same as the previosu rev id as well: A user was added and two user talk pages switched positions, then  that same user was deleted and those same user talk pages switched backed to their orignal positions, exact self revert. The only difference is when the original and last edit was (different time yet same content and entries as original/last entry for previous rev id)

*  rev id: 876286171 - Again, AvicBot was "Listifying from Category:Candidates for speedy deletion". Allthoguh the edits seem to be somewhat related, it doesn't seem like a revert. 4 to 6 to 4 entries (same 4 entries that the original edit started with) 
*  rev id: 876287145 - This has the same edit summary as above, and while a user was added to this category at the time (6 to 7 entries), the revision that reverted it from 3 to 4 entries. the revisions in between removed three users (1 that was added in the orignal edit)

*  rev id: 876287797 - This has the same edit summary as above, and while a user was added to this category (one page switch positions) at the time (7 to 8 entries), the revision that reverted it from 3 to 4 entries. this revisions removed 4 users (a page being removed before the original edit and being added again )

*  rev id: 876289121 - This has the same edit summary as above, and 5 entries were removed during the original edit (lef twith 3 entries), the revision that reverted it from 3 to 4 entries. the revisions in between removed four users (1 that was added and swicthed in the orignal edit)

*  rev id: 876289513 - This has the same idt summary as above, where a user was added to hte original edit (3 to 4 entries). the revision that reverted it removed a user that was added by revision in between, back to the original 3 entries

*  rev id: 876292591 - This has the same edit summary as above, these edits don't seem to be reverts, though. the original edit start with 4 entries and then a catergory was added, but the revision that reverted this removed a user that was added by revision in between (not the category that was added before), but not back to its original edit

*  rev id: 876293533 - This has the same edit summary as above. the original edit start with 5 entries and then a page was added and a couple of pages switched positions; the revision that reverted this removed the page that was added earlier, but not back to its original edit

*  rev id: 876294173 - This has the same edit summary as above, these edits don't seem to be reverts, though. the original edit start with 6 entries and then two pages were removed and a page switched positions; the revision that reverted this removed the page that was added by another revision, but not back to its original edit

*  rev id: 876300057 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; this revert is just showed when the page was back to 0 entries, 16 revisions added and removed

*  rev id: 876308853 - Again, AvicBot was "Listifying from Category:Candidates for speedy deletion". a page was added to the original edit and the revision that reverted it removed 4 entries that were added by revisions in between it (not the user that was added to the original edit earlier)

*  rev id: 876243659 - This edit summary was the same as the one above. another page was added to the user the edit ABOVE, and the reviison that reverted it removed 4 entries (including the user page that was added earlier) 

NOTES:

* Although all the revision id's here are different, as expected, some of the revert ids are the same, even thought the revision id is different - this is because all of those revisions have the same revert id, they are apart of revisions that happened in between reverts that weren't exact. 

* looked at these pages and are no longer in use as of July 2020

In [14]:
#all of the revisions that AvicBot reverted
sdf_another[sdf_another.user_text == "AvicBot"]

rev_timestamp   page_id     rev_id prev_rev_id  is_minor user_text  \
39369       1546329320  47909226  876268182   876267825      True   AvicBot   
42291       1546332061  52628506  876271234   876254649      True   AvicBot   
46746       1546335627  36629985  876275873   876270563      True   AvicBot   
48192       1546336830  36104494  876277467   876270552      True   AvicBot   
59800       1546345236  47909226  876289513   876289121      True   AvicBot   
...                ...       ...        ...         ...       ...       ...   
4666915     1548929121  47909226  881084332   881083925      True   AvicBot   
4669393     1548930948  52628506  881086886   881086473      True   AvicBot   
4672135     1548932749  52628506  881089704   881088771      True   AvicBot   
4674767     1548934542  52628506  881092461   881092021      True   AvicBot   
4690708     1548944721  47909226  881109148   881108609      True   AvicBot   

          user_id seconds_to_prev  curr_bytes delta_bytes  ...  damaging_pred  \
39369    11952314             296         606         -33  ...       0.021028   
42291    11952314           14111           0         -25  ...       0.040676   
46746    11952314            4190      116862         -37  ...       0.011994   
48192    11952314            5404      116862         -37  ...       0.011994   
59800    11952314             290         106          23  ...       0.008412   
...           ...             ...         ...         ...  ...            ...   
4666915  11952314             288          21         -18  ...       0.021911   
4669393  11952314             293           0         -18  ...       0.028396   
4672135  11952314             605           0         -39  ...       0.020855   
4674767  11952314             270           0        -473  ...       0.053051   
4690708  11952314             295           0         -20  ...       0.059977   

         goodfaith_pred  model_version  user_is_bot  user_is_trusted  \
39369          0.984608          0.4.0         True             True   
42291          0.978785          0.4.0         True             True   
46746          0.992761          0.4.0         True             True   
48192          0.992761          0.4.0         True             True   
59800          0.994823          0.4.0         True             True   
...                 ...            ...          ...              ...   
4666915        0.990799          0.5.0         True             True   
4669393        0.987399          0.5.0         True             True   
4672135        0.991229          0.5.0         True             True   
4674767        0.968017          0.5.0         True             True   
4690708        0.987755          0.5.0         True             True   

         user_edit_count page_rev_count page_namespace is_page_redirect day  
39369            1080840          58822              2                0   1  
42291            1080851          11856              2                0   1  
46746            1080860           2637              2                0   1  
48192            1080865           2428              2                0   1  
59800            1080888          58822              2                0   1  
...                  ...            ...            ...              ...  ..  
4666915          1089013          58822              2                0  31  
4669393          1089017          11856              2                0  31  
4672135          1089022          11856              2                0  31  
4674767          1089032          11856              2                0  31  
4690708          1089063          58822              2                0  31  

[313 rows x 30 columns]

In [15]:
len(bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"])

532

In [16]:
len(sdf_another[sdf_another.user_text == "AvicBot"]) 

313

In [17]:
avicbot = df[df.user_text == "AvicBot"]

In [18]:
avicbot23 = sdf_bots[sdf_bots.user_text == "AvicBot"] #using the original data, wanted to print all of AvicBot's revisions made in 3 weeks that were reverted 
avicbot23 = avicbot23[avicbot23.is_self_reverted == True] #then I wanted to see how many of those were self reverted

In [20]:
avicbot23[avicbot23.damaging_pred > 0.09]

rev_timestamp   page_id     rev_id prev_rev_id  is_minor user_text  \
310458      1546475518  52628506  876549388   876545360      True   AvicBot   
1356162     1547074859  52628506  877638210   877637651      True   AvicBot   
2603620     1547763350  52628506  878938875   878937739      True   AvicBot   

          user_id seconds_to_prev  curr_bytes delta_bytes  ...  damaging_pred  \
310458   11952314            1857          65        -451  ...       0.115168   
1356162  11952314             312          46        -388  ...       0.092731   
2603620  11952314             606          26        -252  ...       0.091902   

         goodfaith_pred  model_version  user_is_bot  user_is_trusted  \
310458         0.938539          0.4.0         True             True   
1356162        0.955530          0.5.0         True             True   
2603620        0.959058          0.5.0         True             True   

         user_edit_count page_rev_count page_namespace is_page_redirect day  
310458           1081329          11856              2                0   3  
1356162          1083311          11856              2                0   9  
2603620          1085619          11856              2                0  17  

[3 rows x 30 columns]

In [26]:
#python didn't have a mean function, so i just created one
def mean(list):
    for row in list:
        sum = 0
        sum += row
    avg = sum/(len(list))
    print(avg)      

In [22]:
mean(avicbot23.damaging_pred)

9.622259239739857e-06


In [23]:
mean(avicbot23.goodfaith_pred)

0.0018737866324398417


In [24]:
avicbot23[avicbot23.goodfaith_pred < 0.5]

Empty DataFrame
Columns: [rev_timestamp, page_id, rev_id, prev_rev_id, is_minor, user_text, user_id, seconds_to_prev, curr_bytes, delta_bytes, has_edit_summary, is_reverted, is_revert, is_reverted_to_by_other, is_self_reverted, is_self_revert, revert_target_id, revert_set_size, revert_id, seconds_to_revert, damaging_pred, goodfaith_pred, model_version, user_is_bot, user_is_trusted, user_edit_count, page_rev_count, page_namespace, is_page_redirect, day]
Index: []

[0 rows x 30 columns]

In [25]:
t = avicbot23[avicbot23.damaging_pred > 0.05]
t[t.day<22]
t[t.is_reverted] # w/o, still include 'none'
t = t[t.seconds_to_revert.astype('str').astype('int')<86400]

In [27]:
list = []
for row in t.itertuples():
    for line in Acivbot_self_reverts.itertuples():
        if row.rev_id == line.rev_id:
            list.append(row.rev_id)
            print(f"Rev ID: {line.rev_id}, Revert ID: {line.revert_id}")
        
list

Rev ID: 876254649, Revert ID: 876271234
Rev ID: 876549388, Revert ID: 876564599
Rev ID: 877638210, Revert ID: 877638714
Rev ID: 877978756, Revert ID: 877979328
Rev ID: 878938875, Revert ID: 878940822


[876254649, 876549388, 877638210, 877978756, 878938875]

In [28]:
#checking if i already loooked at these
first20 = bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"].head(20)
done = []
for row in t.itertuples():
    for line in first20.itertuples():
        if row.rev_id == line.rev_id:
            done.append(row.rev_id)
        
done

[876254649]

THESE ARE THE REVISIONS MADE BY AVICBOT, WITH RELATIVELY HIGH DAMAGING PREDICTION ORES SCORES!
==


* rev id: 876549388 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; this edit started with two entries (before 17) and ended up with 0, AVicBot removed these 2  entries in between

* rev id: 877638210 - This edit summary was the same as the previous one, yet the revisions don't even seem to be reverts, this  edit started with 1 entry (before 13) and the revision that "reverted" it, removed this 1 entry  

*  rev id: 877978756 - This edit summary was the same as the previous one, yet the revisions don't even seem to be reverts, this  edit started with 1 entry (before 12) and the revision that "reverted" it, removed this 1 entry  

*  rev id: 878938875 - This edit summary was the same as the previous one, yet the revisions don't even seem to be reverts, this  edit started with 1 entry (before 9) and the revision that "reverted" it, removed this 1 entry

*  rev id: 876254649 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg" was added (couple edits ago) and is now being removed, 19 to 1 ("original" edit) to 0

Why are the damaging predictions for this so high compared to the others? Most of the "reverts" here seem to be AvicBot removing 10+ entries, bringign the page to 0, but isn't this a good thing? So are these scores based on how many are removed or the person removed? Does the criteria for the scores change from bot to bot since they all do different tasks? is there a a bot ORES and user ORES or are they all the same?

In [29]:
avicbot23_namespaces = avicbot23[avicbot23.user_is_bot].groupby('page_namespace')

namespace_df_avicbot = avicbot23_namespaces.agg({
    'rev_id': len,
    'is_reverted': np.sum,
}).rename(columns={
    'rev_id': 'n_revisions',
    'is_reverted': 'n_reverted',
})
namespace_df_avicbot['pct_reverted'] = namespace_df_avicbot.n_reverted / namespace_df_avicbot.n_revisions

namespace_df_avicbot


n_revisions  n_reverted  pct_reverted
page_namespace                                       
2                       532         532           1.0

In [30]:
bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"].sample(10)

rev_id user_text  revert_id  rev_id_reverter user_text_reverter
2640  877568508   AvicBot  877574318        877574318            AvicBot
2964  877839696   AvicBot  877843114        877843114            AvicBot
260   876293533   AvicBot  876295082        876295082            AvicBot
3469  878246609   AvicBot  878248457        878248457            AvicBot
4624  879067917   AvicBot  879100186        879100186            AvicBot
2396  877394615   AvicBot  877395748        877395748            AvicBot
4202  878839526   AvicBot  878841370        878841370            AvicBot
2296  877313318   AvicBot  877316819        877316819            AvicBot
4420  878933361   AvicBot  878934693        878934693            AvicBot
1946  877015029   AvicBot  877024944        877024944            AvicBot

*  rev id: 877568508 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; Avicbot removed two entries leaving the original edit with just one entry...a couple of edits later, and the edit that reverted this, removed all entries

*  rev id: 877839696 - This has the same edit summary as above. Avicbot removed 1 entry leaving the original edit with 20 entries...a couple of edits later, and the edit that reverted this, removed all entries

*  (DONE ALREADY) rev id: 876293533 - This has the same edit summary as above. the original edit start with 5 entries and then a page was added and a couple of pages switched positions; the revision that reverted this removed the page that was added earlier, but not back to its original edit

* rev id: 878246609 - This has the same edit summary as above. Avicbot added 1 entry leaving the original edit with 1 entry...a couple of edits later, and the edit that reverted this, removed all entries

* rev id: 879067917 - This edit summary was the same as the previous one, yet the revisions don't seem to be related and don't even seem to be reverts... 

*  rev id: 877394615 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty)

*  rev id: 878839526 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty), 9 to 13 to 0 entries

*  rev id: 877313318 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 13 to 19 to 0

*  rev id: 878933361 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 19 to 1 to 0

*  rev id: 877015029 - This edit summary was "Listifying from Category:Candidates for speedy deletion". A user was added and that same user was deleted, exact self revert

Questions: 
==

* Why are there adminstrator newsletters on this talk page? (only of months April, May, and June, and sent by MediaWiki message delivery 
* Wonder how people are notified of these discussions, and if users can request the results of trials or wants to give feedback to a bot/point out a mistake on one of these discussions that could change the result of the overall discussion?
* Why does a bot decide to swtich positions of user talk pages? what is the order based on? 
* I know its a robot but how is AvicBot removing and adding candidates within 5-10 minutes? even 30 secs? would not expect these things to happen so quickly since a discussion is involved
